### PR TITLE
Fix critical bug in im2col/col2im convolution implementation

### PR DESCRIPTION
**Pull request by vijay-x-Raj**
## 🐛 Problem
The `im2col_indices` and `col2im_indices` functions had **incorrect column indexing** causing:
- Wrong convolution outputs
- Broken gradient backpropagation
- CNNs unable to learn properly

## ✅ Solution
Fixed the indexing logic from `y * Fw + x::Fh * Fw` to `col_idx * C:(col_idx + 1) * C` ensuring:
- Correct patch extraction
- Proper gradient accumulation
- Mathematically accurate convolutions

## 📝 Changes
- Fixed `im2col_indices()` in `pydeepflow/model.py`
- Fixed `col2im_indices()` in `pydeepflow/model.py`
- Removed duplicate imports
- Added input validation to `ConvLayer.forward()`
- Added comprehensive test suite with 6 tests (all passing)
- Numerical gradient check confirms correctness (error < 1e-3)

## 🧪 Testing
```bash
python test_conv_fix.py

### CODE REVIEW

**Comment by vijay-x-Raj** [comment]
All the merge conflicts have been resolved @ravin-d-27

**Comment by ravin-d-27** [comment]
Thank you ! @vijay-x-Raj
